src/Resources/app/storefront/src/checkout/checkout.plugin.js

```diff
@@ -182,14 +182,17 @@ export default class CheckoutPlugin extends Plugin {
 
     hideStateData(method) {
         let element = $(`[data-adyen-payment-method=${adyenConfiguration.paymentMethodTypeHandlers[method]}]`);
-        if (method === adyenCheckoutOptions.statedataPaymentMethod) {
-            //The state data stored matches this method, show update details button
-            element.find('[data-adyen-payment-container]').hide();
-            element.find('[data-adyen-update-payment-details]').show();
-        } else if (method === 'oneclick' && adyenCheckoutOptions.statedataPaymentMethod === 'storedPaymentMethods') {
-            //The state data stored matches storedPaymentMethods, show update details button for oneclick
-            $(`[data-adyen-payment-method=${adyenConfiguration.paymentMethodTypeHandlers["oneclick"]}]`).find('[data-adyen-payment-container]').hide();
-            $(`[data-adyen-payment-method=${adyenConfiguration.paymentMethodTypeHandlers["oneclick"]}]`).find('[data-adyen-update-payment-details]').show();
+        let selectedPaymentMethod = this.getSelectedPaymentMethodKey();
+        if (method === selectedPaymentMethod) {
+            if (method == 'oneclick') {
+                //The state data stored matches storedPaymentMethods, show update details button for oneclick
```

Comment:
since we are no more returning `storedPaymentMethod` as the type for oneclick, the method and selectedPaymentMethod will match, like all the others.